src/Microsoft.AspNetCore.Mvc.TagHelpers/EnvironmentTagHelper.cs

```diff
@@ -68,27 +92,75 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 return;
             }
 
-            var tokenizer = new StringTokenizer(Names, NameSeparator);
+            var hasExcludeEnvironments = false;
+            if (Exclude != null)
+            {
+                var tokenizer = new StringTokenizer(Exclude, ExcludeEnvironmentsSeparator);
+                foreach (var item in tokenizer)
+                {
+                    var environment = item.Trim();
+                    if (environment.HasValue && environment.Length > 0)
+                    {
+                        hasExcludeEnvironments = true;
+                        if (environment.Equals(currentEnvironmentName))
+                        {
+                            output.SuppressOutput();
+                            return;
+                        }
+                    }
+                }
+            }
+
             var hasEnvironments = false;
```

Comment:
Just need one `hasIncludeEnvironments` variable for both `Names` and `Includes`.